teos-common/src/lightning/messages.rs

```diff
@@ -0,0 +1,355 @@
+    pub encrypted_blob: Vec<u8>,
+}
+
+/// The tracker data message sent by the tower when the requested appointment has been acted upon.
+#[derive(Debug)]
+pub struct TrackerData {
+    pub dispute_txid: Txid,
+    pub penalty_txid: Txid,
+    pub penalty_rawtx: Vec<u8>,
+}
+
+/// The appointment not found message sent by the tower in response to a get appointment message
+/// whose locator didn't match any known appointment.
+#[derive(Debug)]
+pub struct AppointmentNotFound {
+    pub locator: Locator,
+}
+
+/// The get subscription info message (a TEOS custom message, not a bolt13 one).
+#[derive(Debug)]
+pub struct GetSubscriptionInfo {
+    pub signature: String,
+}
+
+/// The subscription info message sent by the tower in response to get subscription info message.
+#[derive(Debug)]
+pub struct SubscriptionInfo {
+    pub available_slots: u32,
+    pub subscription_expiry: u32,
+    // Sent as a TLV. Defaults to an empty vector.
+    pub locators: Vec<Locator>,
+}
+
+impl_writeable_msg!(Register, {
+    pubkey,
+    appointment_slots,
+    subscription_period
+}, {});
+
+impl_writeable_msg!(SubscriptionDetails, {
+    appointment_max_size,
+    start_block,
+    amount_msat,
+}, {
+    // Use `opt_str` and not `opt` to avoid writing a length prefix for strings
+    // since it's already written in the length part of the TLV.
+    (1, invoice, opt_str),
+    (3, signature, opt_str),
+});
+
+impl_writeable_msg!(AddUpdateAppointment, {
+    locator,
+    encrypted_blob,
+    signature,
+}, {
+    (1, to_self_delay, opt),
+});
+
+impl_writeable_msg!(AppointmentAccepted, {
+    locator,
+    start_block,
+}, {
+    // Use `opt_str` and not `opt` to avoid writing a length prefix for strings
+    // since it's already written in the length part of the TLV.
+    (1, receipt_signature, opt_str),
+});
+
+impl_writeable_msg!(AppointmentRejected, {
+    locator,
+    rcode,
+    reason,
+}, {});
+
+impl_writeable_msg!(GetAppointment, {
+    locator,
+    signature,
+}, {});
+
+impl_writeable_msg!(AppointmentData, {
+    locator,
+    encrypted_blob,
+}, {});
+
+impl_writeable_msg!(TrackerData, {
+    dispute_txid,
+    penalty_txid,
+    penalty_rawtx,
+}, {});
+
+impl_writeable_msg!(AppointmentNotFound, {
+    locator,
+}, {});
+
+impl_writeable_msg!(GetSubscriptionInfo, {
+    signature,
+}, {});
+
+impl_writeable_msg!(SubscriptionInfo, {
+    available_slots,
+    subscription_expiry,
+}, {
+    (1, locators, vec)
+});
+
+set_msg_type!(Register, 48848);
+set_msg_type!(SubscriptionDetails, 48850);
+set_msg_type!(AddUpdateAppointment, 48852);
+set_msg_type!(AppointmentAccepted, 48854);
+set_msg_type!(AppointmentRejected, 48856);
+set_msg_type!(GetAppointment, 48858);
+set_msg_type!(AppointmentData, 48860);
+set_msg_type!(TrackerData, 48862);
+set_msg_type!(AppointmentNotFound, 48864);
+// Let these messages get odd types since they are auxiliary messages.
+set_msg_type!(GetSubscriptionInfo, 48865);
+set_msg_type!(SubscriptionInfo, 48867);
```

Comment:
Is there any reasoning behind the values taken for the message types? (apart from them being part of the custom message range and being even/odd depending on whether they are required or not)?

Reply:
Not specifically, just random numbers as placeholders.